### PR TITLE
maint: re-structure to src layout

### PR DESCRIPTION
**Pull request by steven-murray**
This puts the package in a src directory, and also removes the setup.py and setup.cfg as they are no longer needed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,11 +2,92 @@
...
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "21cmSense"
+description = "Compute sensitivies of radio interferometers to the cosmic 21cm signal."
+readme = "README.rst"
+authors = [
+  {name="Jonathan Pober", email="jonathon_pober@brown.edu"},
+  {name="Steven Murray", email="murray.steveng@gmail.com"},
+  {name="Matt Kolopanis", email="matthew.kolopanis@asu.edu"},
```
Comment (mkolopanis) [inline pyproject.toml:12]: actually matthew.kolopanis@gmail.com would be better here for me please.

File: setup.py
Comment (mkolopanis) [inline setup.py:1]: watch one day 21cmsense will need an extension module and we'll have to bring this file back :laughing:

**Review by mkolopanis — COMMENTED** [review]
easy peasy, but one change to my email please.

File: setup.py
Comment (steven-murray) [reply setup.py:1]: lol hopefully by then pyproject will support them ;-)

File: pyproject.toml (hunk 1 above)
Comment (steven-murray) [reply pyproject.toml]: done! And what good luck that you get to have your name @ gmail without numbers or weird gymnastics!

Comment (mkolopanis) [reply pyproject.toml]: the advantages of having a fairly unique name.

**Review by mkolopanis — APPROVED** [review]

**Pull request merged**